src/test/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilderTest.java

```diff
@@ -158,6 +159,28 @@ public void testBuildJnlpFromYamlWithNullEnv() throws Exception {
         assertThat(jnlp.get().getEnv(), hasSize(greaterThan(0)));
     }
 
+    @Test
+    @Issue("JENKINS-71639")
+    public void testInjectPspSecurityContextInJnlp() throws Exception {
+        cloud.setPspSecurityContext(true);
+        PodTemplate template = new PodTemplate();
+        template.setYaml(loadYamlFile("pod-busybox.yaml"));
+        setupStubs();
+        Pod pod = new PodTemplateBuilder(template, slave).build();
+        Map<String, Container> containers = toContainerMap(pod);
+        assertTrue(containers.containsKey("jnlp"));
+
+        Container jnlp = containers.get("jnlp");
+        assertNotNull(jnlp.getSecurityContext());
```

Comment:
Fine enough, but more interesting would be a Kind-based test that actually enables PSA and verifies that you can run a build using this mode.

Reply:
I'll give it a try and see if I can manage to come up with such a test in a reasonable time.

Reply:
I was having a look at the tests in the project, and for now there is nothing around automated tests using `kind` right? Or did I miss something?

Reply:
(Without Kind, there would be no interesting test coverage in this plugin at all.)